### PR TITLE
ENH: stats: add geometric zscore

### PR DESCRIPTION
**Pull request by qbarthelemy**
#### Reference issue
Closes issue #14816.

#### What does this implement/fix?
It implements `gzscore`, the geometric version of zscore, and it adds tests.

#### Additional information
All details can be found in issue #14816.

### CODE REVIEW

**Review by tupui — CHANGES_REQUESTED** [review]
Thanks @qbarthelemy, a really good start! I have a few comments.

I did not yet look at the tests. They seem complete, but I am not sure yet if we need that much since it's calling `zscore` which is well tested.

**Review by tupui — COMMENTED** [review]
Thanks for the update.

**Review by tupui — CHANGES_REQUESTED** [review]

**Review by tupui — APPROVED** [review]
LGTM, you still have a merge conflict before we can merge. @tylerjereddy do you have other comments?

**Review by tupui — APPROVED** [review]
LGTM, thanks @qbarthelemy (CI failures are not related). I will wait a bit in case Tyler or someone else has something to say and merge early next week otherwise.

**Comment by tupui** [comment]
Close-reopen for CI. I will merge if green

**Comment by tupui** [comment]
Thanks @qbarthelemy!

File: scipy/stats/mstats.py
Comment (mdhaber) [reply scipy/stats/mstats.py]: I think that was the correct interpretation.

**Pull request merged**
